valhalla/midgard/sequence.h

```diff
@@ -433,6 +433,10 @@ template <class T> class sequence {
     return memmap.size() + write_buffer.size();
   }
 
+#if defined(__GNUC__) && !defined(__clang__)
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wdeprecated-copy"
```

Comment:
could you post the compiler warning for this just so we can reason about what is going on here

Reply:
The osx compiler in CI does not recognize `-Wdeprecated-copy` and so bails here. The if-def dance is avoid this on osx. https://app.circleci.com/pipelines/github/valhalla/valhalla/5844/workflows/1a663e84-beef-4ef3-8283-b5017c0d261d/jobs/39679

The warning about the deprecated copy comes from
```
/root/project/src/mjolnir/graphbuilder.cc:257:58: warning: implicitly-declared 'constexpr valhalla::midgard::sequence<valhalla::mjolnir::Node>::iterator::iterator(const valhalla::midgard::sequence<valhalla::mjolnir::Node>::iterator&)' is deprecated [-Wdeprecated-copy]
  257 |           sequence<Node>::iterator element = --nodes.end();
      |                                                          ^
In file included from /root/project/valhalla/baldr/graphreader.h:20,
                 from /root/project/src/mjolnir/graphbuilder.cc:14:
```
https://app.circleci.com/pipelines/github/valhalla/valhalla/5842/workflows/fd3c3901-38a9-490c-880a-06ecbec2ea5e/jobs/39674

Reply:
instead of this `ifdef` dance we could just explicitly implement the copy constructor. would you mind if we tried that out?

Reply:
maybe even this will work?

```c++
iterator(const iterator&) = default;
```

Reply:
Seems like that may work :+1: